src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -170,13 +174,13 @@ module.exports = function<T, P, I, TI, PI, C, CX>(config : HostConfig<T, P, I, T
 
     deferredUpdates,
 
-    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | I | TI | null) {
+    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | PI | null) {
       const root : FiberRoot = (container.stateNode : any);
       const containerFiber = root.current;
       if (!containerFiber.child) {
         return null;
       }
-      return containerFiber.child.stateNode;
+      return getPublicInstance(containerFiber.child.stateNode);
```

Comment:
This will become more clear after https://github.com/facebook/react/pull/8545 because then the `stateNode` will have a proper type and Flow would've disallowed passing it to `getPublicInstance`.